sqlite/fields_test.go

```diff
@@ -0,0 +1,59 @@
+package sqlite
+
+import (
+	"math/big"
+	"testing"
+)
+
+func strToPtr(s string) *string {
+	res := new(string)
+	*res = s
+	return res
+}
+
+func TestBigIntToPadded128BitsStr(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    *big.Int
+		expected *string
+	}{
+		{
+			name:     "case small",
+			input:    big.NewInt(123456),
+			expected: strToPtr("0000000000000000000000000001e240"),
+		},
+		{
+			name:     "case zero",
+			input:    big.NewInt(0),
+			expected: strToPtr("00000000000000000000000000000000"),
+		},
+		{
+			name:     "case very large",
+			input:    new(big.Int).Exp(big.NewInt(10), big.NewInt(26), nil),
+			expected: strToPtr("000000000052b7d2dcc80cd2e4000000"),
+		},
+		{
+			name:     "case max",
```

Comment:
please add a test with a negative number

Reply:
That is undefined behaviour and should not be attempted. Currently, we don't provide negative values but only convert unsigned data.